### ISSUE TITLE
Null Modem Mode

### ISSUE THREAD

**Issue by ssshake** [ssshake/esp8266_modem#21]
Add config option to swap rx and tx pins in software

**Comment by kapfab** [comment]
As this is to avoid using a null-modem adapter, that would also imply swapping RTS and CTS, isn’t it? 
By the way, are DTR and DSR wired on the modem (v4, at least)?

**Comment by ssshake** [comment]
Yup that's the idea, request by a customer. I think you're right, hadn't thought about that. It would be easy to swap both pairs.

I'm not sure if this should be a config setting that should get saved or not. I suppose so, if someone can't connect I could have the 'hold flash to reset' button also reset the pin assignment back to default.

**Comment by kapfab** [comment]
Yes, this should be saved or else one would need to carry a null-modem adapter in order to configure the modem so it doesn’t need one.

And this should be cleared with the flash reset in the same way as the 300 bauds setting.

**Comment by kapfab** [comment]
Even better, would be nice to switch behavior just by pressing the flash button between 3 and 4 seconds for example. No one would never need a null-modem adapter anymore.

**Comment by ssshake** [comment]
If using uart flow control as you brought up in https://github.com/ssshake/vintage-computer-wifi-modem/issues/22 would that mean we wouldn't be able to swap RTS and CTS? 

It might eliminate this feature which might be OK.

This feature could still swap RX and TX and be used for troubleshooting in case someone is unsure about what's going on.

Another thing to consider is while this feature is turned on, communication over the USB port will not work. I doubt many people use the device this way though.
